sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -443,6 +443,26 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val FILE_STREAM_SINK_LOG_DELETE = SQLConfigBuilder("spark.sql.streaming.fileSink.log.deletion")
+    .internal()
+    .doc("Whether to delete the expired log files in file stream sink.")
+    .booleanConf
+    .createWithDefault(true)
+
+  val FILE_STREAM_SINK_LOG_COMPACT_LEN =
+    SQLConfigBuilder("spark.sql.streaming.fileSink.log.compactLen")
+      .internal()
+      .doc("Every how many log files is a compaction triggered.")
```

Comment:
Number of log files after which all the previous files are compacted into the next log file.